src/coupons/coupons.controller.ts

```diff
@@ -10,13 +12,19 @@ import {
   Post,
   Patch as Put,
 } from '@nestjs/common';
-import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import {
+  ApiBearerAuth,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
 import { CouponsService } from './coupons.service';
 import { CreateCouponDto } from './dto/create-coupon.dto';
 import { UpdateCouponDto } from './dto/update-coupon.dto';
 
 @Controller({ path: 'coupons', version: '1' })
 @ApiTags('coupons')
+@ApiBearerAuth()
```

Comment:
why this controller can `@ApiBearerAuth()` before export class not before endpoint but other controllers are in class and before endpoint, just asking btw.

Reply:
Reference to [Docs](https://docs.nestjs.com/openapi/decorators) `@ApiBearerAuth()` is both class and method decorator. When you apply it to class-level, it will apply to all method (endpoint) in class.

Reply:
So if its the same things and you use `@Role` for most of the endpoint why wouldn't you put it all class-level

Reply:
Because in another controller some endpoints (like [this](https://meeorder.github.io/meeorder-backend/#/orders/OrdersController_createOrder)) it does not require to use authentication. `@ApiBearerAuth()` will apply to endpoint that require authentication. In `CouponController` it requires authentication in every methods (endpoints) so I can put the `@ApiBearerAuth()` in class level.